teps/add-variable-retries-and-retry-count.md

```diff
@@ -0,0 +1,105 @@
+---
+title: add-variable-retries-and-retry-count
+authors:
+  - "@zhangyumeng"
+creation-date: 2020-10-14
+last-updated: 2020-10-19
+status: proposed
+---
+
+# TEP-0026: Add Variable `retries` and `retry-count`
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+- [Alternatives](#Alternatives)
+- [References](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+This TEP proposes adding `retries` and current retry times to Pipeline variables.
+
+## Motivation
+
+PipelineTask can be rerun after failure when user specifies `retries` in PipelineTask. In that case, exposing the retry count can give user more information. User can dispatch a notification after certain times of failure, or customize logs according to retry count.
+
+### Goals
+
+- add 2 variables indicating retries and retry count of a PipelineTask
+
+### Non-Goals
+
+- execute different PipelineTask based on retry count
+- emit an event on every retry
+
+## Proposal
+
+There are 2 variable substitutions `tasks.<taskName>.retries` and `tasks.<taskName>.retry-count` needed to be supported:
```

Comment:
@pritidesai just wanted to bring this to your attention cuz it feels related to adding status info https://github.com/tektoncd/community/blob/master/teps/0028-task-execution-status-at-runtime.md#proposal

feels like `tasks.<taskName>.retries`  and `tasks.<taskName>.status` are nicely aligned? (unless we wanted to consider `retries` part of the overall status info, e.g. something like `tasks.<taskName>.status.retries`?)

Reply:
thanks @bobcatfish

> feels like tasks.<taskName>.retries and tasks.<taskName>.status are nicely aligned?
yup `tasks.<taskName>.retries` fits well with `tasks.<taskName>.status`.

I like your suggestion of going with `context` to avoid exposing `retry` count of a task to other tasks.